addons/stock/models/stock_quant.py

```diff
@@ -524,12 +524,16 @@ def _merge_quants(self):
         query = """WITH
                         dupes AS (
                             SELECT min(id) as to_update_quant_id,
-                                (array_agg(id ORDER BY id))[2:array_length(array_agg(id), 1)] as to_delete_quant_ids,
+                                (array_agg(id ORDER BY id)) as to_delete_quant_ids,
```

Comment:
Doesn't seems correct. 
The `to_update_quant_id` will part of `to_delete_quant_ids`, then the `_up AS` CTE will updated a row that will be deleted in the last statements.

![Screenshot_20210628_162513](https://user-images.githubusercontent.com/53175387/123654945-dba43880-d82e-11eb-8b23-c999d09b174f.png)

Reply:
I notice that there isn't any test on it. It should be because it is a critical part of the stock. We will backport a test from master `test_inventory_7`

Reply:
Oh OK. I'll wait for the backport of the test and fix it.

Reply:
```python3
    def test_merge_quant(self):
        """ Check that duplicated quants are correctly update.
        """
        vals = {
            'product_id': self.product1.id,
            'product_uom_id': self.uom_unit.id,
            'location_id': self.stock_location.id,
            'quantity': 1,
        }
        self.env['stock.quant'].create(vals)
        self.env['stock.quant'].create(vals)
        self.assertEqual(len(self.env['stock.quant']._gather(self.product1, self.stock_location)), 2.0)
        self.assertEqual(self.env['stock.quant']._get_available_quantity(self.product1, self.stock_location), 2.0)
        self.env['stock.quant']._quant_tasks()
        quant = self.env['stock.quant'].search([
            ('location_id', '=', self.stock_location.id),
            ('product_id', '=', self.product1.id)
        ])
        self.assertEqual(len(quant), 1)
        self.assertEqual(quant.quantity, 2)
```
Add this in your branch and remove it in the last PR of the forwardbot (master)